kafka/common/pkg/kafka/consumer_factory.go

```diff
@@ -87,7 +93,20 @@ func (c kafkaConsumerGroupFactoryImpl) StartConsumerGroup(groupID string, topics
 }
 
 func NewConsumerGroupFactory(client sarama.Client) KafkaConsumerGroupFactory {
-	return kafkaConsumerGroupFactoryImpl{client}
+	return kafkaConsumerGroupFactoryImpl{client: client, clientCloner: cloneClient}
+}
+
+func cloneClient(client sarama.Client) (sarama.Client, error) {
```

Comment:
why do we need to _clone_ the client? :thinking:

Reply:
https://github.com/knative/eventing-contrib/pull/1165#issuecomment-619786596